stock_internal_transfer_transit/stock_internal_transfer.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014 initOS GmbH & Co. KG (<http://www.initos.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+from openerp.osv import orm
+from openerp.netsvc import LocalService
+
+
+class StockInternalTransfer(orm.TransientModel):
+    _inherit = 'stock.internal.transfer'
+
+    def _prepare_picking(self, cr, uid, ids, context=None):
+        assert len(ids) == 1,\
+            'This function should only be used for a single id at a time.'
+
+        picking_data = super(StockInternalTransfer, self).\
+            _prepare_picking(cr, uid, ids, context=context)
+        wiz = self.browse(cr, uid, ids[0], context=context)
+
+        no_transit = context and \
+            context.get('stock_internal_transfer_no_transit', False)
```

Comment:
I didn't see the set of this context.
Maybe, the transit location, can be set as an optional parameter ?
If the transit location is set, you create 2 pickings, else, you create only on picking.

What do you think ?